pytoda/preprocessing/crawlers.py

```diff
@@ -127,3 +131,58 @@ def get_smiles_from_pubchem(
                 logger.warninig(f'Did not find any result for drug: {drug}')
                 return []
             continue
+
+
+def remove_pubchem_smiles(smiles_list: List[str]) -> List:
+    """
+    Function for removing PubChem molecules from a list of smiles.
+
+    Args:
+        smiles_list (List): List of SMILES strings.
+
+    Returns:
+        List:  Filtered list of SMILES, all SMILES pointing to PubChem
+            molecules are removed.
+    """
+
+    if not isinstance(smiles_list, list):
+        raise TypeError(f'Please pass list, not {type(smiles_list)}')
+
+    canonicalizer = Canonicalization(sanitize=False)
+
+    filtered = list(filter(lambda x: not query_pubchem(x), smiles_list))
+    # Canonicalize molecules and filter again (sanity check)
+    filtered_canonical = list(
+        filter(lambda x: not query_pubchem(canonicalizer(x)), filtered)
+    )
+    return filtered_canonical
+
+
+def query_pubchem(
+    smiles: str,
+    return_id: bool = False,
+) -> Union[bool, Tuple[bool, int]]:
```

Comment:
let's try to avoid conditional return types.
I add a suggestion returning a Tuple only, and an additional function `is_pubchem` selecting the boolean value

Reply:
and then use `is_pubchem` in `remove_pubchem_smiles` instead.

Reply:
good suggestion, I implemented this.